spec/system/admin/users_system_spec.rb

```diff
@@ -37,5 +37,23 @@
 
       expect(page.find(".alert")).to have_content "Deleted that user"
     end
+
+    it "filters users by name" do
+      new_user = User.create!(name: "Willow", email: "email@email.com", password: "blahblah!")
+      user_names = User.all.pluck(:name)
+
+      visit admin_users_path
+      user_names.each do |name|
+        expect(page).to have_content(name)
+      end
+
+      user_names.delete("Willow")
+
+      fill_in "filterrific_search_name", with: new_user.name
+      user_names.each do |name|
+        expect(page.find("table")).not_to have_content(name)
+      end
```

Comment:
This seems odd... `User.all` should include the new user as well. The fact that both this expectation and thenext one both pass seems like a bug somewhere.

In general I wouldn't rely on randomly generated information here. I'd just flat out delete all users and create two new ones (the one you want and the one you don't) and use those to test. Who's to say one of the randomly generated users won't have "willow" in its name somewhere?

Reply:
Thanks for the feedback! This setup was weird. I think it's better now though!

I just noticed that users aren't randomly generated, they're set in the rails_helper here https://github.com/rubyforgood/human-essentials/blob/main/spec/rails_helper.rb#L160-L162. 

So I think this approach is reliable/safe [c10bb55](https://github.com/rubyforgood/human-essentials/pull/3393/commits/c10bb553a44cf624744709fb6aea566c8cca6437?diff=split&w=0). 

It grabs all the user names, expects the names to be on the page, puts the first user's name in the search input field, expects the page to contain only the first user's name. This approach also passes locally for me!

Side note... I did try the delete all users and set up new ones approach, but the specs still failed.